File: data/multilang_comparison/program_cpp_vector/src/func44.cpp
#include "program_cpp_vector.hpp" 
Array* func44(Array_param* vars, int loopsFactor) {
   size_t pCounter = vars->size;
   Array* array21;
   if (pCounter > 0) {
      array21 = vars->data.at(--pCounter);
      array21->refC++;
      DEBUG_COPY(array21->id);
   } else {
      array21 = new Array();
      array21->size = 87;
      array21->refC = 1;
      array21->id = 21;
      array21->data = std::vector<unsigned int>(array21->size, 0);
      DEBUG_NEW(array21->id);
   }
   return array21;
}

File: data/multilang_comparison/program_go/src/func44.go
package main
import "math/rand"
var _ = rand.Intn
func Func44(vars *ArrayParam, loopsFactor int) *Array{
   var pCounter int = vars.Size;
   _ = pCounter
   var array21 *Array
   if pCounter > 0 {
       array21 = vars.Data[pCounter-1]
       pCounter--
       array21.RefC++
   } else {
       array21 = &Array{
           Size: 87,
           RefC: 1,
           Id: 21,
           Data: make([]uint32, 87),
       }
   }
   return array21;
}

